Modules/_elementtree.c

```diff
@@ -968,13 +965,13 @@ element_setstate_from_attributes(ElementObject *self,
     Py_INCREF(tag);
     Py_XSETREF(self->tag, tag);
 
-    _clear_joined_ptr(&self->text);
-    self->text = text ? JOIN_SET(text, PyList_CheckExact(text)) : Py_None;
-    Py_INCREF(JOIN_OBJ(self->text));
+    new_val = text ? JOIN_SET(text, PyList_CheckExact(text)) : Py_None;
```

Comment:
You can reuse the `text` variable for the new value. The old value no longer used.